spec/engine/miq_ae_method_service/miq_ae_service_model_base_spec.rb

```diff
@@ -84,6 +84,14 @@
         expect(test_class.name).to eq('MiqAeMethodService::MiqAeServiceMiqAeServiceModelSpec_TestVmOrTemplate')
         expect(test_class.superclass.name).to eq('MiqAeMethodService::MiqAeServiceVmOrTemplate')
       end
+
+      it 'does not list private attributes' do
+        expect(MiqAeMethodService::MiqAeServiceModelBase).not_to(receive(:expose).with('properties'))
+        obj     = MiqAeServiceModelSpec::TestPrivateAttrExpose.create
+        svc_obj = described_class.model_name_from_active_record_model(MiqAeServiceModelSpec::TestPrivateAttrExpose).constantize.find(obj.id)
+        expect(svc_obj.methods.include?(:properties)).to(eq(false))
+        obj.destroy
```

Comment:
Tests are run within database transactions, you do not need to cleanup the objects as part of the test.